nixos/modules/services/backup/sanoid.nix

```diff
@@ -199,6 +199,13 @@ in
       after = [ "zfs.target" ];
       startAt = cfg.interval;
     };
+
+    # Put those packages in the global environment
+    # so that syncoid can find them when targeting this host through ssh.
+    environment.systemPackages = with pkgs; [
```

Comment:
I guess this is mostly harmless if they aren't used, but I'm not sure it makes sense to assume anyone running sanoid will also be using the machine as a syncoid target. Personally, I only add these packages for a single user on my target machines, just because it seems tidier.

Reply:
Yes, you're both right, and IIRC `syncoid` outputs a warning message when those (quite small) utilities are missing (`lzop`, `mbuffer`), yet I think it's quite easy to forget about installing them. Hosts running `sanoid` are likely to be the targets of tools using them (be it `syncoid` or another one), that's why I think it's a good thing to made them available.
AFAIU having those utilities installed in `sanoid`'s package wrapper or `syncoid`'s package wrapper, would not help. Those utilities are called remotely by the local `syncoid`.

```console
$ nix path-info -rsSh $(which mbuffer) $(which lzop)
/nix/store/567zfi9026lp2q6v97vwn640rv6i3n4c-libunistring-1.1  	   1.8M	   1.8M
/nix/store/4563gldw8ibz76f1a3x69zq3a1vhdpz9-libidn2-2.3.4     	 350.4K	   2.1M
/nix/store/jd99cyc0251p0i5y69w8mqjcai8mcq7h-xgcc-12.2.0-libgcc	 139.3K	 139.3K
/nix/store/46m4xx889wlhsdj72j38fnlyyvvvvbyb-glibc-2.37-8      	  28.8M	  31.1M
/nix/store/4iabmjjq95069myjsrid8pk2ib3yz4nn-openssl-3.0.10    	   6.2M	  37.3M
/nix/store/sxrh37b22ga4q42kcapfh1746jk887ky-lzo-2.10          	 392.3K	  31.5M
/nix/store/6k4g5zdbddw4w72ahna4sz2fbb091xbr-lzop-1.04         	 234.8K	  31.7M
/nix/store/ffj3cdbza79b1lgd5z0ylg76fj0byk5w-mbuffer-20230301  	 114.2K	  37.4M
```